Sources/ToosieSlide/UICollectionViewCarouselLayout.swift

```diff
@@ -79,40 +100,51 @@ open class UICollectionViewCarouselLayout: UICollectionViewFlowLayout {
     updateSectionInsets()
   }
 
-  open override func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
-    /// Since `invalidateLayout()` is not called everytime the collection view `.bounds` changes, just listen
+  open override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
+    /// Since `invalidateLayout()` is not called every time the collection view `.bounds` changes, just listen
     /// and invalidate layout by caching the collection view size. `layoutAttributesForElements(in rect: CGRect)` is called
     /// every time the collection view doesn't know/isn't "sure" about where to place cells. It is called upon rotation as well.
     if latestKnownCollectionViewSize != collectionView?.bounds.size {
       latestKnownCollectionViewSize = collectionView?.bounds.size
-      invalidateLayout()
+      return true
     }
-    return super.layoutAttributesForElements(in: rect)
+    return false
   }
   
   open override func targetContentOffset(
     forProposedContentOffset proposedContentOffset: CGPoint,
     withScrollingVelocity velocity: CGPoint
   ) -> CGPoint {
-    // The current offset
-    let currentDistance = CGFloat(currentVisibleCell) * (itemSize.width + minimumLineSpacing)
+    // check if fast enough to scroll
+    guard let collectionView = collectionView, abs(velocity.x) > lowestVelocitySensitivity else {
+      return CGPoint(x: currentOffset, y: 0)
+    }
+    
+    let futureCellIndex = velocity.x > 0 ?
+    min(currentVisibleCellIndex + 1, collectionView.numberOfItems(inSection: 0) - 1) :
```

Comment:
same as above

Reply:
in teoria no, perche se è vuota non potresti manco scrollare, ma comunque ho aggiunto il guard su ✅